src/transformers/pipelines/__init__.py

```diff
@@ -530,9 +577,24 @@ def pipeline(
         task = get_task(model, use_auth_token)
 
     # Retrieve the task
-    targeted_task, task_options = check_task(task)
-    if pipeline_class is None:
-        pipeline_class = targeted_task["impl"]
+    if task in custom_tasks:
+        targeted_task, task_options = clean_custom_task(custom_tasks[task])
+        if pipeline_class is None:
+            if not trust_remote_code:
+                raise ValueError(
+                    f"Loading this pipeline requires you to execute the code in the pipeline file in that"
+                    f" repo on your local machine. Make sure you have read the code there to avoid malicious use, then"
+                    f" set the option `trust_remote_code=True` to remove this error."
+                )
+            class_ref = targeted_task["impl"]
+            module_file, class_name = class_ref.split(".")
+            pipeline_class = get_class_from_dynamic_module(
+                model, module_file + ".py", class_name, revision=revision, use_auth_token=use_auth_token
+            )
```

Comment:
Reading the code, I understand that as soon as the config defines `custom_pipelines` then the `pipeline(..)` will necessarily attempt to read the remote code, even if the user doesn't want that and really want to run the original pipeline.

What happens if I really want to run the pipeline and do not trust the remote code and just wants to use the classical code ? 
Or maybe since `transformers` has added the implementation and so we don't want to `trust_remote_code` anymore ?

Maybe we could use `trust_remote_code=None` as indecise (current behavior), `=True` download the remote code. `=False` don't download remote code, and just keep on running.

Wdyt ?

Reply:
Will see what I can do! Sounds like a reasonable plan.